### PR TITLE
Add edit events ld

### PR DESCRIPTION
**Pull request by LoganDemaray316**
Adds the Edit Events popup that can be used for Juries and Recitals. A start and end time can be passed with intervals that can make a list of buttons for that times the student can choose from.

Edited Files:
EventComponent.vue

Added Files:
EventItemEdit.vue

### CODE REVIEW

File: src/components/EventItemEdit.vue [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+          <v-col>
+            <v-card-subtitle class="font-weight-bold text-darkGray">
+              Musical Selection
+            </v-card-subtitle>
+            <v-card-text>
+              <v-row>
+                <v-col cols="6">
+                  <v-text-field
+                    label="Search your repertoire"
+                    variant="solo"></v-text-field>
```
Comment (ethanimooney) [inline src/components/EventItemEdit.vue:111]: Look at the Vuetify docs for Text Fields, particularly 'Append Icon' to get the search icon here.

File: src/components/EventItemEdit.vue [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+            start_time: "13:00:00",
+            end_time: "15:00:00",
+            interval: 10,
+          },
+        ],
+      };
+    },
+    computed: {},
+    methods: {
+      timesToResults(start_time, end_time, interval) {
```
Comment (ethanimooney) [inline src/components/EventItemEdit.vue:246]:
```
function getTimeSlots(startDate, endDate, interval) {
  var slots = [];

  var intervalMillis = interval * 60 * 1000;

  while (startDate < endDate) {
    var mins = (startDate.getMinutes() + "0").slice(0, 2);
    slots.push(startDate.getHours() + ":" + mins);
    startDate.setTime(startDate.getTime() + intervalMillis);
  }

  return slots;
}
```

Code above does the same thing, just using the JS Date datatype. We should be able to parse our database call into Dates so this seems more efficient if you want to change it.

File: src/components/EventItemEdit.vue [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+                  size="x-small"
+                  class="buttonGradient text-white font-weight-bold">
+                  Add to repertoire
+                </v-btn>
+              </v-col>
+            </v-row>
+            <v-row>
+              <v-col>
+                <v-card class="eventsGradient fullBorderCurve mainblur"
+                  >//Addition Selections Go Here</v-card
```
Comment (ethanimooney) [inline src/components/EventItemEdit.vue:179]: Some typos/random text here?

File: src/components/EventItemEdit.vue [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+                </v-col>
+              </v-row>
+              <v-row>
+                <v-col>
+                  <v-card class="eventsGradient fullBorderCurve mainblur">
+                    <v-card-title class="font-weight-bold text-white pl-5">
+                      Selected
+                    </v-card-title>
+                    <v-card-text>
+                      <v-card class="fullBorderCurve">
```
Comment (ethanimooney) [inline src/components/EventItemEdit.vue:130]: Same here, can use `rounded-lg` instead of that class.

File: src/components/EventItemEdit.vue [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+                    elevation="0"
+                    rounded="pill"
+                    class="buttonGradient text-white font-weight-bold">
+                    Add New
+                  </v-btn>
+                </v-col>
+              </v-row>
+              <v-row>
+                <v-col>
+                  <v-card class="eventsGradient fullBorderCurve mainblur">
```
Comment (ethanimooney) [inline src/components/EventItemEdit.vue:125]: Same here, can use `rounded-lg` instead of that class.

**Review by ethanimooney — CHANGES_REQUESTED** [review]

File: src/components/EventItemEdit.vue [hunk 6]
```diff
@@ -0,0 +1,283 @@
+<template>
+  <v-container fluid class="eventComponent">
+    <v-card class="fullBorderCurve mainblur">
```
[the remark this answers is not in the document]
Comment (LoganDemaray316) [reply src/components/EventItemEdit.vue:3]: Fixed

File: src/components/EventItemEdit.vue (hunk 1 above)
Comment (LoganDemaray316) [reply src/components/EventItemEdit.vue:111]: The append icon had the effect of placing the icon outside of the text field. It is in this push

File: src/components/EventItemEdit.vue (hunk 3 above)
Comment (LoganDemaray316) [reply src/components/EventItemEdit.vue:179]: Removed. There is nothing in here yet.

File: src/components/EventItemEdit.vue (hunk 4 above)
Comment (LoganDemaray316) [reply src/components/EventItemEdit.vue:130]: Done

File: src/components/EventItemEdit.vue (hunk 5 above)
Comment (LoganDemaray316) [reply src/components/EventItemEdit.vue:125]: Done

File: src/components/EventItemEdit.vue (hunk 2 above)
Comment (LoganDemaray316) [reply src/components/EventItemEdit.vue:246]: I'm not sure if I need to enter in different data, but I get an error saying it cannot use .getMinutes()

Comment (LoganDemaray316) [reply src/components/EventItemEdit.vue:246]: Uncaught (in promise) TypeError: startDate.getMinutes is not a function
    at Proxy.getTimeSlots (EventItemEdit.vue:252:33)
    at EventItemEdit.vue:67:41
    at Proxy.renderFnWithContext (runtime-core.esm-bundler.js:868:19)

Comment (ethanimooney) [reply src/components/EventItemEdit.vue:246]: You just need to make sure `startDate, endDate` are JS Date datatypes. You may need to do some parsing prior to passing them into make sure they are.

File: src/components/EventItemEdit.vue (hunk 1 above)
Comment (ethanimooney) [reply src/components/EventItemEdit.vue]: Weird - I can't pull this to look right now cause I am not at a point where I can switch branches. We can just leave it for now and deal with it in a later PR.

File: src/components/EventItemEdit.vue [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+        var slots = [];
+
+        var intervalMillis = interval * 60 * 1000;
+
+        while (startDate < endDate) {
+          var mins = (startDate.getMinutes() + "0").slice(0, 2);
+          slots.push(startDate.getHours() + ":" + mins);
+          startDate.setTime(startDate.getTime() + intervalMillis);
+        }
+        console.log(slots);
```
Comment (ethanimooney) [inline src/components/EventItemEdit.vue:257]: Can remove this

**Review by ethanimooney — APPROVED** [review]
LGTM, just remove the one comment

**Pull request merged**